Migrate from Circle CI to GitHub Actions

colincasey opened the pull request:
[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

Malax commented on integration-test/openjdk-11/exception/test.sh:
```diff
@@ -8,16 +8,16 @@ function_invocation_response_file="$(mktemp)"
 
 pushd "${test_dir}"
-./mvnw compile
+./mvnw --no-transfer-progress compile
```
We probably want `--batch` here, which includes `--no-transfer-progress`.

Malax commented on pom.xml:
```diff
@@ -290,6 +290,7 @@
                 <version>3.4.1</version>
                 <configuration>
                     <source>${maven.compiler.source}</source>
+                    <doclint>all,-syntax</doclint>
```
Is this the same issue as with the SDK where `javadoc` picks up an annotation as a doctag?

colincasey replied:
yes.  and i can't figure out why or reproduce it 😞

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

Malax commented on pom.xml:
```diff
@@ -288,9 +288,6 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
                 <version>3.4.1</version>
-                <configuration>
-                    <source>${maven.compiler.source}</source>
-                </configuration>
```
I assume this was removed because it's the default anyway?

Malax reviewed (approved):
Thanks for taking care of this! Appreciated! 👍🏻
